src/main/java/it/gov/pagopa/initiative/statistics/service/BaseStatisticsEvaluationService.java

```diff
@@ -99,7 +99,16 @@ private void evaluatePartitionRecords(int partition, List<ConsumerRecord<String,
                 // skipping retry messages scheduled by other application
                 .filter(this::isNotRetry)
                 // transforming the record2entity stream into a pair record2initiativeBased stream
-                .flatMap(r2e -> toInitiativeBasedEntityStream(r2e.getValue()).map(i -> Triple.of(r2e.getKey(), getInitiativeId(i), i)))
+                .flatMap(r2e -> {
+                    try {
+                        return toInitiativeBasedEntityStream(r2e.getValue()).map(i -> Triple.of(r2e.getKey(), getInitiativeId(i), i));
+                    } catch (Exception e) {
+                        errorRecords.add(Triple.of(r2e.getKey(),
+                            "[INITIATIVE_STATISTICS_EVALUATION][%s] Unexpected error: %s".formatted(getFlowName(), r2e.getValue()),
+                            e));
+                    return null;
```

Comment:
you cannot return null inside a flatMap!
It should always be a Mono or Flux...

We could return Mono.empty() (or Flux) in order to simply ignoring the case, we already have logged it anyway)